### ISSUE TITLE
The wrong codes about llamagen model with SJD

### ISSUE THREAD

**Issue by Haodong-Lei-Ray** [tyshiwo1/Accelerating-T2I-AR-with-SJD#3]
Hi, I notice a bug it is there:
https://github.com/tyshiwo1/Accelerating-T2I-AR-with-SJD/blob/b389cfb2b6aa228efab32190f98bc13c3fd4b8e6/scheduler/jacobi_iteration_lumina_mgpt.py#L1069

LlamaGen's Solver chooses the "jacobi_iteration_lumina_mgpt.JacobiSampler" as new Solver class. 
But the logits_processor of LlamaGen is always without the attributes "w_latent_dim". So  img_width always equals zero.

Therefore, even if the type of token initation "multi_token_init_scheme" is "repeat" or "sample" type, then it finally turns to "random" type, because there is the condition "img_width>0". 

![Image](https://github.com/user-attachments/assets/d8e18c4c-3379-4110-adf2-12d285f0239c)

Additionally, I pay attention to your "comment" as "# TODO: this is for mgpt, the first 3 tokens <start, h, w>". So if the model is LlamaGen, why do not you change it? Is it wrong that prefill_num + 3?

Thank for your reply.
